src/net/java/textilej/parser/markup/textile/phrase/SimpleTextilePhraseModifier.java

```diff
@@ -49,7 +49,15 @@ public SimpleTextilePhraseModifier(String delimiter, SpanType spanType, boolean
 	@Override
 	protected String getPattern(int groupOffset) {
 		String quotedDelimiter = quoteLite(getDelimiter());
-		
+
+		// special casing to avoid breaking freenet keys:
+		if (this.delimiter == "-") {
```

Comment:
Strings have to compared using `.equals()` (except for some very constrained circumstances which do not apply here), so `delimiter.equals("-")` would be correct. If `delimiter` can only be `null` you’d have to check for that, too, because (obviously) you can’t call `.equals()` on `null`.